### PR TITLE
chore: bump go to 1.18

### PR DESCRIPTION
**Pull request by cmwaters**

### CODE REVIEW

**Review by mark-rushakoff — APPROVED** [review]
I'm not sure why the linter started complaining about these existing lines -- maybe the new `indirect` entries in go.mod updated a version of a linter unexpectedly?

+1 to moving to go1.18 anyway.

**Comment by cmwaters** [comment]
```
level=warning msg="[linters context] gosimple is disabled because of go1.18. You can track the evolution of the go1.18 support by following the https://github.com/golangci/golangci-lint/issues/2649."
  level=warning msg="[linters context] staticcheck is disabled because of go1.18. You can track the evolution of the go1.18 support by following the https://github.com/golangci/golangci-lint/issues/2649."
  level=warning msg="[linters context] structcheck is disabled because of go1.18. You can track the evolution of the go1.18 support by following the https://github.com/golangci/golangci-lint/issues/26[49](https://github.com/tendermint/tendermint/runs/7612082007?check_suite_focus=true#step:5:51)."
  level=warning msg="[linters context] stylecheck is disabled because of go1.18. You can track the evolution of the go1.18 support by following the https://github.com/golangci/golangci-lint/issues/2649."
  level=warning msg="[linters context] unused is disabled because of go1.18. You can track the evolution of the go1.18 support by following the https://github.com/golangci/golangci-lint/issues/2649."
```
Seems like a lot of the linting stuff was deprecated in go1.18

**Comment by mark-rushakoff** [comment]
We are on v1.47.2 of golangci-lint, and v1.47.3 was released two hours ago, including https://github.com/golangci/golangci-lint/pull/3001 to "remove some go1.18 limitations". Shouldn't hurt to update to that newer patch release.

File: .golangci.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 linters:
   enable:
     - asciicheck
-    - bodyclose
+    # - bodyclose // to be fixed by golangci-lint
```
Comment (mark-rushakoff) [inline .golangci.yml:4]: bodyclose support is supposed to be re-enabled as part of golangci-lint v1.47.3, according to https://github.com/golangci/golangci-lint/pull/3001.

**Review by mark-rushakoff — APPROVED** [review]
New updates still LGTM.

**Pull request merged**